### PR TITLE
No heavy ds/v1

### PR DESCRIPTION
**Pull request by cryogenian**
Removed `HeavyweightDatasource` and made `QueryAssociate` an alias for `Kleisli` (it seems that it used to be `Kleisli` once, due to `mapF` and `mapK` defined)

### CODE REVIEW

**Review by rintcius — COMMENTED** [review]
Inline comments suggest to keep it non-breaking for the plugin datasources, but we could also just break it so that we don't have to keep the adapter around.

File: impl/src/main/scala/quasar/impl/datasources/DefaultDatasources.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import quasar.api.datasource.DatasourceError._
 import quasar.api.resource._
 import quasar.connector.datasource.Reconfiguration
 import quasar.contrib.scalaz.MonadError_
-import quasar.impl.{CachedGetter, IndexedSemaphore, QuasarDatasource, ResourceManager}, CachedGetter.Signal._
+import quasar.impl.{CachedGetter, IndexedSemaphore, ResourceManager, QuasarDatasource}, CachedGetter.Signal._
```
Comment (wemrysi) [inline impl/src/main/scala/quasar/impl/datasources/DefaultDatasources.scala:27]: Much better 😆

File: impl/src/main/scala/quasar/impl/evaluate/QueryFederator.scala [hunk 2]
```diff
@@ -48,22 +48,27 @@ import shims.{orderToCats, monadToScalaz}
   * request given their current capabilities.
   */
 object QueryFederator {
-  def apply[T[_[_]]: BirecursiveT, F[_]: Monad: MonadResourceErr, G[_], H[_], R, P <: ResourcePathType](
-      sources: AFile => F[Option[Source[QuasarDatasource[T, G, H, R, P]]]])
-      : Kleisli[F, (T[QScriptEducated[T, ?]], Option[Offset]), FederatedQuery[T, QueryAssociate[T, G, R]]] =
-    Kleisli(new QueryFederatorImpl(sources).tupled)
+
+  def apply[T[_[_]]: BirecursiveT]: PartiallyApplied[T] = new PartiallyApplied[T]
```
Comment (wemrysi) [inline impl/src/main/scala/quasar/impl/evaluate/QueryFederator.scala:52]: It is more ergonomic (in scala-2, at least) to push all the constraints to `PartiallyApplied#apply`, doing so should allow syntax like

```scala
QueryFederator[Fix] { file =>
  ...
}
```

without having to explicitly invoke `.apply`.

**Review by wemrysi — APPROVED** [review]
Looks good, great to see the elimination of the recursive type everywhere!

File: impl/src/main/scala/quasar/impl/package.scala [hunk 3]
```diff
@@ -33,4 +36,10 @@ package object impl {
 
   type MonadQuasarErr[F[_]] = MonadError_[F, QuasarError]
   def MonadQuasarErr[F[_]](implicit ev: MonadQuasarErr[F]): MonadQuasarErr[F] = ev
+
+
+  private [impl] type QuasarDatasource[F[_], G[_], R, P <: ResourcePathType] =
+    Datasource[F, G, InterpretedRead[ResourcePath], R, P]
+
+  private [impl] type DatasourceModule = LightweightDatasourceModule
```
[the remark this answers is not in the document]
Comment (wemrysi) [reply impl/src/main/scala/quasar/impl/package.scala:44]: 👍 to just having `DatasourceModule`. I'm fine with breaking it here or if we want to have a separate PR just for the breaking change.

File: impl/src/main/scala/quasar/impl/evaluate/QueryFederator.scala (hunk 2 above)
Comment (cryogenian) [reply impl/src/main/scala/quasar/impl/evaluate/QueryFederator.scala:52]: Oh, that makes sense, thank you!

**Comment by cryogenian** [comment]
@rintcius You're right, there is no reason why we need to keep `LightweightDatasource` at all. I'll remove/rename it, anyways the change in plugins would be literally `LightweightDatasourceFoo -> DatasourceFoo`.

File: impl/src/main/scala/quasar/impl/Quasar.scala [hunk 4]
```diff
@@ -97,7 +97,7 @@ object Quasar extends Logging {
       schedulerRefs: IndexedStore[F, UUID, SchedulerRef[Json]],
       pushes: PrefixStore.SCodec[F, UUID :: ResourcePath :: HNil, ∃[Push[?, SqlQuery]]],
       offsets: Store[F, UUID :: ResourcePath :: HNil, ∃[OffsetKey.Actual]],
-      queryFederation: QueryFederation[Fix, Resource[F, ?], QueryAssociate[Fix, Resource[F, ?], EvalResult[F]], R],
```
[the remark this answers is not in the document]
Comment (rintcius) [reply impl/src/main/scala/quasar/impl/Quasar.scala:100]: Wondering... is `Fix` the only use of `T[_[_]` in QueryFederation / FederatedQuery / MimirQueryFederation(Impl) / MimirConnector / ... ? If so, can we get rid of the `T[_[_]` here too?

[the remark this answers is not in the document]
Comment (wemrysi) [reply impl/src/main/scala/quasar/impl/Quasar.scala:100]: Are you suggesting specializing to `Fix` everywhere?

[the remark this answers is not in the document]
Comment (rintcius) [reply impl/src/main/scala/quasar/impl/Quasar.scala:100]: Yes

[the remark this answers is not in the document]
Comment (cryogenian) [reply impl/src/main/scala/quasar/impl/Quasar.scala:100]: :+1: For fixing it in `QueryFederation`, not sure about fixing it in `FederatedQuery`, should be OK too I suppose

[the remark this answers is not in the document]
Comment (wemrysi) [reply impl/src/main/scala/quasar/impl/Quasar.scala:100]: Yeah, I think fixing in `QueryFederation` is ok, I'd prefer the other types to remain parametric.

**Review by rintcius — APPROVED** [review]

**Pull request merged**